### PR TITLE
chore: added test cases for Logs

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

Added test cases for Logs:
- Flows covered:
  - logs Explorer page - (single/multiple queries & search and qb views)
  - logs List view
  - logs table view
  - toolbaractions - qb, search, histogram

#### Related Issues / PR's

| Before | After |
|---|---|
| Screenshot 2024-04-09 at 8 44 24 PM | Screenshot 2024-04-09 at 8 35 03 PM |

### CODE REVIEW

File: frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+		// switch to table view
+		// eslint-disable-next-line sonarjs/no-duplicate-string
+		await userEvent.click(queryByTestId('table-view') as HTMLElement);
+
+		expect(
+			queryByText(
+				'{"container_id":"container_id","container_name":"container_name","driver":"driver","eta":"2m0s","location":"frontend","log_level":"INFO","message":"Dispatch successful","service":"frontend","span_id":"span_id","trace_id":"span_id"}',
+			),
+		);
```
Comment (Vikrant2520) [inline frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx:109]: expect(queryByText("fbhf")) ? assertion seems missing here

File: frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		(useGetExplorerQueryRange as jest.Mock).mockReturnValue({
+			data: { payload: logsQueryRangeSuccessNewFormatResponse },
+			isLoading: false,
+			isFetching: false,
+			isError: true,
+		});
+		const { queryByText, queryByTestId } = renderer();
+
+		expect(
+			queryByText('Something went wrong. Please try again or contact support.'),
```
Comment (Vikrant2520) [inline frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx:141]: assertions ? should not rely on default assertion here

File: frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+		);
+
+		// switch to table view
+		await userEvent.click(queryByTestId('table-view') as HTMLElement);
+
+		expect(
+			queryByText('Something went wrong. Please try again or contact support.'),
```
Comment (Vikrant2520) [inline frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx:148]: same here

File: frontend/src/pages/LogsExplorer/__tests__/LogsExplorer.test.tsx [hunk 4]
```diff
@@ -144,4 +159,102 @@ describe('Logs Explorer Tests', () => {
...
+								</VirtuosoMockContext.Provider>
+							</QueryBuilderProvider>
+						</MockQueryClientProvider>
+					</I18nextProvider>
+				</Provider>
+			</MemoryRouter>,
+		);
+
+		const histogramToggle = container.querySelector('button[role="switch"]');
+		expect(histogramToggle).toBeChecked();
```
Comment (Vikrant2520) [inline frontend/src/pages/LogsExplorer/__tests__/LogsExplorer.test.tsx:255]: histogram toggle has already been tested in the first test of this file , can remove this

File: frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx (hunk 1 above)
Comment (SagarRajput-7) [reply frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx:109]: Thanks, these got cleared out somehow is batch editing or something, fixed everywhere

File: frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx (hunk 2 above)
Comment (SagarRajput-7) [reply frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx:141]: Done

File: frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx (hunk 3 above)
Comment (SagarRajput-7) [reply frontend/src/container/LogsExplorerViews/tests/LogsExplorerViews.test.tsx:148]: done

File: frontend/src/pages/LogsExplorer/__tests__/LogsExplorer.test.tsx (hunk 4 above)
Comment (SagarRajput-7) [reply frontend/src/pages/LogsExplorer/__tests__/LogsExplorer.test.tsx]: Removed, thanks

**Review by Vikrant2520 — APPROVED** [review]

**Review by Rajat-Dabade — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Pull request merged**